azure-client-runtime/src/main/java/com/microsoft/azure/OperationStatus.java

```diff
@@ -23,102 +15,28 @@
     private T result;
 
     /**
-     * Create a new OperationStatus from the provided HTTP response.
-     * @param originalHttpRequest The HttpRequest that initiated the long running operation.
-     * @param originalHttpResponse The HttpResponse from the request that initiated the long running
-     *                             operation.
-     * @param serializer The serializer used to deserialize the response body.
-     */
-    OperationStatus(HttpRequest originalHttpRequest, HttpResponse originalHttpResponse, SerializerAdapter<?> serializer) {
-        final int httpStatusCode = originalHttpResponse.statusCode();
-
-        if (httpStatusCode != 200) {
-            final String fullyQualifiedMethodName = originalHttpRequest.callerMethod();
-            final String originalHttpRequestMethod = originalHttpRequest.httpMethod();
-            final String originalHttpRequestUrl = originalHttpRequest.url();
-
-            if (originalHttpRequestMethod.equalsIgnoreCase("PUT") || originalHttpRequestMethod.equalsIgnoreCase("PATCH")) {
-                if (httpStatusCode == 201) {
-                    pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(fullyQualifiedMethodName, originalHttpResponse, originalHttpRequestUrl, serializer);
-                } else if (httpStatusCode == 202) {
-                    pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(fullyQualifiedMethodName, originalHttpResponse, originalHttpRequestUrl, serializer);
-                    if (pollStrategy == null) {
-                        pollStrategy = LocationPollStrategy.tryToCreate(fullyQualifiedMethodName, originalHttpResponse);
-                    }
-                }
-            } else /* if (originalRequestHttpMethod.equalsIgnoreCase("DELETE") || originalRequestHttpMethod.equalsIgnoreCase("POST") */ {
-                if (httpStatusCode == 202) {
-                    pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(fullyQualifiedMethodName, originalHttpResponse, originalHttpRequestUrl, serializer);
-                    if (pollStrategy == null) {
-                        pollStrategy = LocationPollStrategy.tryToCreate(fullyQualifiedMethodName, originalHttpResponse);
-                    }
-                }
-            }
-        }
-    }
-
-    /**
-     * Update the properties of this OperationStatus from the provided response.
-     * @param httpResponse The HttpResponse from the most recent request.
+     * Create a new OperationStatus with the provided PollStrategy.
+     * @param pollStrategy The polling strategy that the OperationStatus will use to check the
+     *                     progress of a long running operation.
      */
-    void updateFrom(HttpResponse httpResponse) throws IOException {
-        pollStrategy.updateFrom(httpResponse);
+    OperationStatus(PollStrategy pollStrategy) {
```

Comment:
How do consumers of this value check the ProvisioningState? I'm looking around the management SDK and I'm finding that many services have their own independent ProvisioningState enum with an associated String value.

When a user receives an OperationStatus value they may want to be able to see e.g. I'm in the CREATING ProvisioningState versus the RESOLVING_DNS ProvisioningState.

Reply:
Sure, that makes sense. I'll create another pull request where I add in a provisioningState() getter.